wrappers/csharp/src/Device.cs

```diff
@@ -154,5 +153,52 @@ public SensorList Sensors
             }
         }
     }
+	
+	public class AdvancedDevice : Device
+    {
+        private const string AdvancedModeDisabledException = "Advanced mode has not been enabled";
+
+        internal AdvancedDevice(IntPtr dev) : base(dev)
+        {
+
+        }
+
+        public bool AdvancedMode
+        {
+            get
+            {
+                int enabled = 0;
+                NativeMethods.rs2_is_enabled(m_instance, out enabled, out object error);
+
+                return enabled == 1 ? true : false;
+            }
+            set
+            {
+                NativeMethods.rs2_toggle_advanced_mode(m_instance, value ? 1 : 0, out object error);
+            }
+        }
+
+        public string JsonConfiguration
+        {
+            get
+            {
+                if (!AdvancedMode)
```

Comment:
I think that we can avoid this validation, doesn't the next line throw an exception with a string something like "Device is not in advanced mode"?

Reply:
Yes, it does. I guess `rs2_load_json` and `rs2_serialize_json` do the same checks. But as far as I can see the whole NET wrapper doesn't do any error handling. No code is making use of the returned error objects.

Reply:
This is where we make use of the error object:
https://github.com/IntelRealSense/librealsense/blob/5d987c41962ea70b42357abc1d2f4038867445a8/wrappers/csharp/src/Helpers.cs#L52-L77

Reply:
Ah, that is actually really nice :) I'll remove the check.